mk/rabbitmq-dist.mk

```diff
@@ -123,23 +162,32 @@ $(DIST_DIR)/%.ez:
 		&& grep -q '^prepare-dist::' $(SRC_DIR)/Makefile) || \
 		$(MAKE) --no-print-directory -C $(SRC_DIR) prepare-dist \
 		APP=$(APP) VSN=$(VSN) EZ_DIR=$(EZ_DIR)
-	$(verbose) (cd $(DIST_DIR) && $(ZIP) $(ZIP_V) -r $*.ez $*)
+	$(verbose) (cd $(DIST_DIR) && $(ZIP) $(ZIP_V) -r $(notdir $(EZ)) $(notdir $(EZ_DIR)))
 	$(verbose) rm -rf $(EZ_DIR)
 
+$(MIX_DIST_EZS): $(mix_task_archive_deps_installed)
+	$(verbose) rm -rf $(DIST_DIR)/tmp
+	$(verbose) mkdir $(DIST_DIR)/tmp
+	$(verbose) cd $(SRC_DIR) && $(MIX) do deps.get, deps.compile, compile, archive.build.all -e -o $(realpath $(DIST_DIR))/tmp
+
+$(mix_task_archive_deps_installed):
+	mix archive.install --force https://github.com/hairyhum/mix_task_archive_deps/releases/download/0.1.0/mix_task_archive_deps-0.1.0.ez
+
 # We need to recurse because the top-level make instance is evaluated
 # before dependencies are downloaded.
 
 MAYBE_APPS_LIST = $(if $(shell test -f $(ERLANG_MK_TMP)/apps.log && echo OK),$(ERLANG_MK_TMP)/apps.log)
 
 dist:: $(ERLANG_MK_RECURSIVE_DEPS_LIST) all
-	$(gen_verbose) $(MAKE) do-dist DIST_PLUGINS_LIST="$(ERLANG_MK_RECURSIVE_DEPS_LIST) $(MAYBE_APPS_LIST)"
+	$(gen_verbose) rm -f $(DIST_DIR)/*.ez
+	$(verbose) $(MAKE) do-dist DIST_PLUGINS_LIST="$(ERLANG_MK_RECURSIVE_DEPS_LIST) $(MAYBE_APPS_LIST)"
 
 test-dist:: $(ERLANG_MK_RECURSIVE_TEST_DEPS_LIST) test-build
 	$(gen_verbose) $(MAKE) do-dist DIST_PLUGINS_LIST="$(ERLANG_MK_RECURSIVE_TEST_DEPS_LIST) $(MAYBE_APPS_LIST)"
 
 do-dist:: $(DIST_EZS)
-	$(verbose) unwanted='$(filter-out $(DIST_EZS),$(wildcard $(DIST_DIR)/*.ez))'; \
-	test -z "$$unwanted" || (echo " RM     $$unwanted" && rm -f $$unwanted)
```

Comment:
`deps.compile` is required to get versions for deps. Mix doesn't report a version for a non-compiled dependency. 
We probably don't need `compile` task for listing dependencies.
We cannot do `deps.get` and `archive.build.all.list` in the same command, because we don't want output from `deps.get`